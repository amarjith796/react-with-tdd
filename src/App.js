import React, { Component } from 'react';
import './App.css';
import TodoView from "./Todo/todo"
import AddTodo from "./Todo/addtodo";
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [{ taskname: 'C', status: true },
      { taskname: 'C++', status: false },
      { taskname: 'Java', status: false },
      { taskname: 'Python', status: false },
      { taskname: 'COBOL', status: false }],
      editText: null
    }
  }

  handleStatus = (name) => {
    let data = [...this.state.data];
    let index = data.findIndex(d => d.taskname === name);
    data[index].status = !data[index].status;
    this.setState({
      data: data
    })
  }

  handleDelete = (obj) => {
    let data = [...this.state.data];
    data = data.filter(d => d.taskname != obj.taskname);
    this.setState({
      data: data
    })
  }

  handleAddTask = (taskname) => {
    let data = [...this.state.data];
    data.push({
      taskname: taskname,
      status: false
    });
    this.setState({
      data: data
    })
  }

  handleEditTodo = (taskname) => {
    this.setState({
      editText: taskname
    })
  }
  handleUpdateTodo = (prevtaskname, taskname) => {
    let data = [...this.state.data];
    let index = data.findIndex(d => d.taskname === prevtaskname);
    data[index].taskname = taskname;
    this.setState({
      data: data,
      editText: null
    })
  }

  render() {
    const { data, editText } = this.state;
    return (
      <div className="App-header">
        <AddTodo handleAddTask={this.handleAddTask} editText={editText} handleUpdateTodo={this.handleUpdateTodo} />
        <TodoView data={data}
          handleStatus={this.handleStatus}
          handleDelete={this.handleDelete}
          handleEditTodo={this.handleEditTodo} />
      </div>
    );
  }
}

export default App;
