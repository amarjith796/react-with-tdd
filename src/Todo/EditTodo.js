import React, { Component } from 'react';
import PropTypes from 'prop-types';
class EditTodo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            textInput: this.props.editText
        }
    }
    handleOnchange = (event) => {
        this.setState({
            textInput: event.target.value
        });
    }
    render() {
        const { editText, handleUpdateTodo } = this.props;
        const { textInput } = this.state;
        return (
            <div style={{ display: 'flex', width: '500px' }}>
                <input type='text' style={{ flex: 1 }} value={textInput} onChange={this.handleOnchange} />
                <button style={{ padding: '10px', background: 'steelblue', border: '1px solid steelblue', cursor: 'pointer' }} onClick={() => { handleUpdateTodo(editText, textInput) }}  >
                    Update
                </button>
            </div>);
    }
}
EditTodo.propTypes = {
    editText: PropTypes.any,
    handleUpdateTodo: PropTypes.func
};

export default EditTodo;