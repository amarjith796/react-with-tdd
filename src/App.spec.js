import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import TodoView from "./Todo/todo";
import AddTodo from "./Todo/addtodo";
import EditTodo from "./Todo/EditTodo";
describe('App', () => {

  let wrapper;
  beforeEach(() => wrapper = shallow(<App />));
  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the TodoView Component', () => {
    expect(wrapper.containsMatchingElement(<TodoView data={wrapper.instance().state.data} handleStatus={wrapper.instance().handleStatus} handleDelete={wrapper.instance().handleDelete} handleEditTodo={wrapper.instance().handleEditTodo} />)).toEqual(true);
  });
  it('should render the AddTodo Component', () => {
    expect(wrapper.containsMatchingElement(<AddTodo handleAddTask={wrapper.instance().handleAddTask} />))
  })
  it('should render the EditTodo Component', () => {
    expect(wrapper.containsMatchingElement(<EditTodo editText={wrapper.instance().state.editText} handleUpdateTodo={wrapper.instance().handleUpdateTodo} />))
  })
});