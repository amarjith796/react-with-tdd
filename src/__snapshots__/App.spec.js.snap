// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AddTodo
          handleAddTask={[Function]}
        />,
        <TodoView
          data={
            Array [
              Object {
                "status": true,
                "taskname": "C",
              },
              Object {
                "status": false,
                "taskname": "C++",
              },
              Object {
                "status": false,
                "taskname": "Java",
              },
              Object {
                "status": false,
                "taskname": "Python",
              },
              Object {
                "status": false,
                "taskname": "COBOL",
              },
            ]
          }
          handleDelete={[Function]}
          handleEditTodo={[Function]}
          handleStatus={[Function]}
        />,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleAddTask": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [
            Object {
              "status": true,
              "taskname": "C",
            },
            Object {
              "status": false,
              "taskname": "C++",
            },
            Object {
              "status": false,
              "taskname": "Java",
            },
            Object {
              "status": false,
              "taskname": "Python",
            },
            Object {
              "status": false,
              "taskname": "COBOL",
            },
          ],
          "handleDelete": [Function],
          "handleEditTodo": [Function],
          "handleStatus": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AddTodo
            handleAddTask={[Function]}
          />,
          <TodoView
            data={
              Array [
                Object {
                  "status": true,
                  "taskname": "C",
                },
                Object {
                  "status": false,
                  "taskname": "C++",
                },
                Object {
                  "status": false,
                  "taskname": "Java",
                },
                Object {
                  "status": false,
                  "taskname": "Python",
                },
                Object {
                  "status": false,
                  "taskname": "COBOL",
                },
              ]
            }
            handleDelete={[Function]}
            handleEditTodo={[Function]}
            handleStatus={[Function]}
          />,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleAddTask": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [
              Object {
                "status": true,
                "taskname": "C",
              },
              Object {
                "status": false,
                "taskname": "C++",
              },
              Object {
                "status": false,
                "taskname": "Java",
              },
              Object {
                "status": false,
                "taskname": "Python",
              },
              Object {
                "status": false,
                "taskname": "COBOL",
              },
            ],
            "handleDelete": [Function],
            "handleEditTodo": [Function],
            "handleStatus": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
